databricks/koalas/indexes.py

```diff
@@ -1494,6 +1494,53 @@ def is_all_dates(self):
         """
         return isinstance(self.spark_type, TimestampType)
 
+    def repeat(self, repeats: int) -> "Index":
+        """
+        Repeat elements of a Index.
+
+        Returns a new Index where each element of the current Index
+        is repeated consecutively a given number of times.
+
+        Parameters
+        ----------
+        repeats : int
+            The number of repetitions for each element. This should be a
+            non-negative integer. Repeating 0 times will return an empty
+            Index.
+
+        Returns
+        -------
+        repeated_index : Index
+            Newly created Index with repeated elements.
+
+        See Also
+        --------
+        Series.repeat : Equivalent function for Series.
+        MultiIndex.repeat : Equivalent function for MultiIndex.
+
+        Examples
+        --------
+        >>> idx = ks.Index(['a', 'b', 'c'])
+        >>> idx
+        Index(['a', 'b', 'c'], dtype='object')
+        >>> idx.repeat(2)
+        Index(['a', 'b', 'c', 'a', 'b', 'c'], dtype='object')
+        """
+        if not isinstance(repeats, int):
+            raise ValueError("`repeats` argument must be integer, but got {}".format(type(repeats)))
+        elif repeats < 0:
+            raise ValueError("negative dimensions are not allowed")
+
+        sdf = self._internal.sdf.select(self._internal.scol)
```

Comment:
Ah, wait. we can integrate with MultiIndex if we

1. `self._internal.scol` -> `self._internal.index_scols`
2. and use `self._internal.index_map` when create `index_map`
?

like this
```python
        if not isinstance(repeats, int):
            raise ValueError("`repeats` argument must be integer, but got {}".format(type(repeats)))
        elif repeats < 0:
            raise ValueError("negative dimensions are not allowed")

        sdf = self._internal.sdf.select(self._internal.index_scols)
        internal = _InternalFrame(
            sdf=sdf, index_map=self._internal.index_map
        )
        kdf = DataFrame(internal)  # type: DataFrame
        if repeats == 0:
            return DataFrame(kdf._internal.with_filter(F.lit(False))).index
        else:
            return ks.concat([kdf] * repeats).index
```
```

Reply:
`scol` in `Index` could be different from `_internal.index_scols[0]`.

Reply:
@ueshin Ah, yeah i just checked Implementation of them. seems we don't need to use `index_scols[0]` for `Index`.